packages/aws-cdk/test/toolkit/cli-io-host.test.ts

```diff
@@ -0,0 +1,351 @@
+    });
+
+    test('applies default style by message level in TTY mode', async () => {
+      const host = new CliIoHost({ useTTY: true });
+      await host.notify({
+        time: new Date(),
+        level: 'warn',
+        action: 'synth',
+        code: 'TEST',
+        message: 'warning message',
+        forceStdout: true,
+      });
+
+      expect(mockStdout).toHaveBeenCalledWith(chalk.yellow('warning message') + '\n');
+    });
+
+    test('does not apply styles in non-TTY mode', async () => {
+      const host = new CliIoHost({ useTTY: false });
+      await host.notify({
+        time: new Date(),
+        level: 'info',
+        action: 'synth',
+        code: 'TEST',
+        message: 'unstyled message',
+        forceStdout: true,
+      });
+
+      expect(mockStdout).toHaveBeenCalledWith('unstyled message\n');
+    });
+  });
+
+  describe('timestamp handling', () => {
+    test('includes timestamp for DEBUG level with gray color', async () => {
+      const host = new CliIoHost({ useTTY: true });
+      const testDate = new Date('2024-01-01T12:34:56');
+
+      await host.notify({
+        time: testDate,
+        level: 'debug',
+        action: 'synth',
+        code: 'TEST',
+        message: 'debug message',
+        forceStdout: true,
+      });
+
+      expect(mockStdout).toHaveBeenCalledWith(`[12:34:56] ${chalk.gray('debug message')}\n`);
+    });
+
+    test('includes timestamp for TRACE level with gray color', async () => {
+      const host = new CliIoHost({ useTTY: true });
+      const testDate = new Date('2024-01-01T12:34:56');
+
+      await host.notify({
+        time: testDate,
+        level: 'trace',
+        action: 'synth',
+        code: 'TEST',
+        message: 'trace message',
+        forceStdout: true,
+      });
+
+      expect(mockStdout).toHaveBeenCalledWith(`[12:34:56] ${chalk.gray('trace message')}\n`);
+    });
+
+    test('excludes timestamp for other levels but includes color', async () => {
+      const host = new CliIoHost({ useTTY: true });
+      const testDate = new Date('2024-01-01T12:34:56');
+
+      await host.notify({
+        time: testDate,
+        level: 'info',
+        action: 'synth',
+        code: 'TEST',
+        message: 'info message',
+        forceStdout: true,
+      });
+
+      expect(mockStdout).toHaveBeenCalledWith(chalk.white('info message') + '\n');
+    });
+  });
+
+  describe('CI mode behavior', () => {
+    test('writes to stdout in CI mode regardless of level', async () => {
+      const host = new CliIoHost({ useTTY: true, ci: true });
+      await host.notify({
+        time: new Date(),
+        level: 'info',
+        action: 'synth',
+        code: 'TEST',
+        message: 'ci message',
+      });
+
+      expect(mockStdout).toHaveBeenCalledWith(chalk.white('ci message') + '\n');
+      expect(mockStderr).not.toHaveBeenCalled();
+    });
+
+    test('writes to stdout in CI mode with forceStdout', async () => {
```

Comment:
log level info + CI mode results in stdout. why does it matter that we're explicitly `forceStdout: true`?

Reply:
i want to see this test have log level error. then the `forceStdout: true` does something

Reply:
https://github.com/aws/aws-cdk/pull/32503/files#diff-f0cfd80588d37b405edd7b83de83795144306f15f5178ffd9f5dd9c5618cfeafR235

Added this test though in practice this logic is pretty irrelevant as it's not invoked. Though this is unconnected and that doesn't exist yet so sure why not